### ISSUE TITLE
Extend behaviour of "setHeader" method

### ISSUE THREAD

**Issue by jkfb** [silkimen/cordova-plugin-advanced-http#24]
Hi guys!

It would be a nice feature to extend the behaviour of the _setHeader_ method to support different headers for all further requests based on the host part of an URL. The URL parameter should be optional to preserve the current behaviour.

Something like:
```
function setHeader() {
        var url = '*';
        var header = arguments[0];
        var value = arguments[1];

        if (arguments.length === 3) {
            var url = getHost(arguments[0]); // get the host part of the URL
            var header = arguments[1];
            var value = arguments[2];
        }

        this.headers[url][header] = value;
}
```

What is your opinion on this?

**Comment by silkimen** [comment]
Do we really need to provide URL here? Isn't it better to provide host part instead?

e.g.
using this `setHeader('google.de', 'myKey', 'myValue')` 
instead of `setHeader('https://www.google.de/?gws_rd=ssl', 'myKey', 'myValue')`

**Comment by silkimen** [comment]
will be included in v1.7.0
